lib/validators/movielist.js

```diff
@@ -0,0 +1,12 @@
+'use strict';
+
+const Joi = require('joi');
+
+module.exports = Joi.object().keys({
+  release_year: Joi.number().integer().min(1878).max(9999),
+  start_year: Joi.number().integer().min(1878).max(9999),
+  end_year: Joi.number().integer().min(1878).max(9999).optional(),
+  name: Joi.string().min(1).max(255),
+  similar: Joi.boolean().optional()
+}).without('release_year', ['start_year', 'end_year']).without('start_year', ['release_year']).without('end_year', ['release_year'])
```

Comment:
Does this mean the filter only works for the following 3 options: 
* release_year only 
* start_year but nothing else 
* end_year but nothing else

Not really familiar with the `Joi` library yet

Reply:
The filter works for those 3 options, as well as specifying both start_year and end_year (a range search for movie)